chain/jsonrpc/src/lib.rs

```diff
@@ -283,16 +288,73 @@ impl JsonRpcHandler {
         .map_err(|_| timeout_err())?
     }
 
+    async fn send_tx(
+        &self,
+        tx: SignedTransaction,
+        check_only: bool,
+    ) -> Result<NetworkClientResponses, RpcError> {
+        Ok(self
+            .client_addr
+            .send(NetworkClientMessages::Transaction {
+                transaction: tx,
+                is_forwarded: false,
+                check_only,
+            })
+            .map_err(|err| RpcError::server_error(Some(ServerError::from(err))))
+            .await?)
+    }
+
+    async fn send_tx_sync(&self, params: Option<Value>) -> Result<Value, RpcError> {
+        self.send_or_check_tx(params, false).await
+    }
+
+    async fn check_tx(&self, params: Option<Value>) -> Result<Value, RpcError> {
+        self.send_or_check_tx(params, true).await
+    }
+
+    async fn send_or_check_tx(
+        &self,
+        params: Option<Value>,
+        check_only: bool,
+    ) -> Result<Value, RpcError> {
+        let tx = parse_tx(params)?;
+        let tx_hash = (&tx.get_hash()).to_base();
+        match self.send_tx(tx, check_only).await? {
+            NetworkClientResponses::ValidTx => {
+                if check_only {
+                    Ok(Value::Null)
+                } else {
+                    jsonify(Ok(Ok(RpcBroadcastTxSyncResponse {
+                        transaction_hash: tx_hash,
+                        is_routed: false,
+                    })))
+                }
+            }
+            NetworkClientResponses::RequestRouted => {
+                if check_only {
+                    Err(RpcError::server_error(Some("Node doesn't track this shard. Cannot determine whether the transaction is valid".to_string())))
```

Comment:
@bowenwang1996  
What does this error mean for user?

Reply:
@fckt Good catch.

We should help users to understand that they are expected to reach another node that tracks the shard.

Reply:
@frol how to better phrase it? Should we include something like "try another node"?

Reply:
I was thinking the node must be found automatically? (maybe the request should contain a sender_id?) How to use this method?